pkg/controller/remotemachineset/remotemachineset_controller.go

```diff
@@ -193,6 +193,11 @@ func (r *ReconcileRemoteMachineSet) syncMachineSets(
 	// we extract installer image refs.
 	var amiID string
 	for _, ms := range remoteMachineSets.Items {
+		if ms.Spec.Template.Spec.ProviderSpec.Value == nil {
+			cdLog.WithField("machineSet", ms.Name).Warn("machine set has nil ProviderSpec")
```

Comment:
Is this something that deserves a warning? It is valid for a machine set to have an empty provider spec. Are we assuming that the user is not allowed to create their own machine sets?

Reply:
This leads me to the concern that this code that searches for the AMI ID may not be resilient enough to user-created machinesets, where the user may use a different or invalid AMI ID.

Reply:
Hive presently assumes full control of machine sets in any Hive managed cluster. In future I could see someone requesting disabling this or supporting user created, but not yet.

As such seeing a machine set with no ProviderSpec is pretty weird and for our purposes, we should probably be getting a warning in our logs.